src/watertap_contrib/seto/costing/units/chemical_softening_0D.py

```diff
@@ -0,0 +1,826 @@
+# from watertap.core.zero_order_costing import ZeroOrderCostingData
+# from idaes.core import declare_process_block_class
+
+from pyomo.environ import (
+    Constraint,
+    Expression,
+    Reals,
+    NonNegativeReals,
+    log,
+    Var,
+    Param,
+    Set,
+    Suffix,
+    value,
+    check_optimal_termination,
+    units as pyunits,
+)
+
+from watertap.costing.util import register_costing_parameter_block
+from idaes.core.util.misc import StrEnum
+from ..util import (
+    make_capital_cost_var,
+    make_fixed_operating_cost_var,
+)
+
+"""
+TODO: Remove the hard coded constants in the costing equations
+      Don't fix variables in the unit model code
+"""
+
+
+def build_chem_softening_cost_param_block(blk):
+    """
+    Parameters and variables to be used in the costing model
+    """
+    costing = blk.parent_block()
+
+    blk.sed_basin_depth = Var(
+        initialize=4.5,
+        bounds=(3, 6),
+        units=pyunits.m,
+        doc="Depth of sedimentation basin",
+    )
+
+    blk.sludge_disposal_cost = Var(
+        initialize=35,
+        units=costing.base_currency / pyunits.tons,
+        doc="Cost of sludge disposal $/tonne",
+    )
+
+    blk.MgCl2_cost = Var(
+        initialize=1.5,
+        units=costing.base_currency / pyunits.kg,
+        doc="Cost of MgCl2 $/kg",
+    )
+
+    blk.Na2CO3_cost = Var(
+        initialize=0.65,
+        units=costing.base_currency / pyunits.kg,
+        doc="Cost of soda ash $/kg",
+    )
+
+    blk.fix_all_vars()
+
+
+@register_costing_parameter_block(
+    build_rule=build_chem_softening_cost_param_block,
+    parameter_block_name="chemical_softening",
+)
+def cost_chem_softening(blk):
+
+    """
+    Capital and operating costs for excess lime and soda softening
+    """
+    chem_soft = blk.costing_package.chemical_softening
+    make_capital_cost_var(blk)
+    make_fixed_operating_cost_var(blk)
+
+    blk.mixer_energy = Var(
```

Comment:
I've been fairly loose on people about this in WaterTAP, but since we are focusing on integration with power systems in this repo, anything with units of Watts should be named `power`, and anything with Joules or kWh should include the name `energy`. So in this case, can we rename the var as `mixer_power`?